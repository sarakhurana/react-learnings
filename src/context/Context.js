import React, { useReducer } from "react";
import rootReducer from "../store/index";

const initialState = {
  post: { posts: [] },
};
export const Context = React.createContext(initialState);

const { Provider } = Context;

export const StateProvider = ({ children, value }) => {
  let [state, dispatch] = useReducer(rootReducer, initialState);
  state = value ? value : state;
  return <Provider value={{ state, dispatch }}>{children}</Provider>;
};
