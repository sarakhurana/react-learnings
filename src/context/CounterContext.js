import React, { useReducer } from "react";
import reducer from "../store/reducer"

const initialState = {
  counter:0,
}
export const CounterContext = React.createContext(initialState);

const { Provider } = CounterContext;

export const StateProvider = ({ children, value }) => {
  let [state, dispatch] = useReducer(reducer, initialState);
  state = value ? value : state;
  return <Provider value={{ state, dispatch }}>{children}</Provider>;
};