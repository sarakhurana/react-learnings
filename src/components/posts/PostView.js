import React, {
  useCallback,
  useContext,
  useEffect,
  useRef,
  useState,
} from "react";
import CreatePost from "../../containers/post/CreatePost";
import { Context } from "../../context/Context";
import "./postView.css";
import { fetchPosts, getPosts } from "../../store/actions";
import ReactModal from "react-modal";
import { Link } from "react-router-dom";
import { getPostService } from "../../services/GetPostsService";

const PostView = () => {
  const { state, dispatch } = useContext(Context);
  const getPostRef = useRef(false);
  const [isCreatePost, setCreatePost] = useState(false);
  const [posts, setPosts] = useState(null);
  const [error, setError] = useState(false);

  const handleCreatePostClick = () => {
    setCreatePost(true);
  };

  const getPosts = useCallback(async () => {
    const { error, result } = await getPostService();
    console.log(result);
    if (!error) setPosts(result);
    else setError(error);
  });

  useEffect(async () => {
    if (!getPostRef.current) {
      getPosts();
      // fetchPosts(dispatch);
      getPostRef.current = true;
    }
  }, [getPostRef]);
  return (
    <div className="post-view-container">
      <div className="header">
        <div className="header-title-elements">
          <div className="lady"></div>
          <div className="title"></div>
          <div className="likes-photo"></div>
        </div>
        <div className="header-create-post">
          <button
            className="btn-create-post"
            data-testid="test-create-button"
            onClick={handleCreatePostClick}
          >
            Create Post
          </button>
        </div>
      </div>
      <ReactModal
        className="modal-container"
        overlayClassName="modal-overlay"
        isOpen={isCreatePost}
        ariaHideApp={false}
        onRequestClose={() => setCreatePost(false)}
      >
        <CreatePost setCreatePost={setCreatePost} isCreatePost={isCreatePost} getPosts={getPosts}/>
      </ReactModal>
      <div className="main-view">
        <div className="post-container">
          {posts &&
            posts.map((post, index) => {
              return (
                <div key={index}>{post.body}</div>
                // <Link
                //   key={index}
                //   className="post-content"
                //   to={{
                //     pathname: `/postview/${post.postId}`,
                //     state: {
                //       postId: post.postId,
                //     },
                //   }}
                // >
                //   {post && (
                //     <>
                //       {post.image && <img
                //         className="post-image"
                //         src={post.image}
                //         alt={post.title}
                //       ></img>}
                //       <span className="post-description">{post.body}</span>
                //     </>
                //   )}
                // </Link>
              );
            })}
        </div>
      </div>
    </div>
  );
};

export default PostView;
