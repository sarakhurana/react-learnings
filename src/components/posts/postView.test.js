import { render } from "@testing-library/react";
import { StateProvider } from "../../context/Context";
import PostView from "./PostView";

jest.mock("../../containers/post/CreatePost", () => () => {
  return (
    <>
      <label>Enter Post</label>
      <input data-testid="test-create-post" />
    </>
  );
});

describe("PostView", () => {
  it("should render", () => {
    const { asFragment } = render(
      <StateProvider>
        <PostView />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should give option to create post", () => {
    const { getByTestId } = render(
      <StateProvider>
        <PostView />
      </StateProvider>
    );
    const element = getByTestId("test-create-post");

    expect(element).toBeInTheDocument();
  });
});
