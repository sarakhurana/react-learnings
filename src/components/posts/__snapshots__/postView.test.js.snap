// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostView should render 1`] = `
<DocumentFragment>
  <div
    class="post-view-container"
  >
    <label>
      Enter Post
    </label>
    <input
      data-testid="test-create-post"
    />
    <div
      class="post-container"
    />
  </div>
</DocumentFragment>
`;
